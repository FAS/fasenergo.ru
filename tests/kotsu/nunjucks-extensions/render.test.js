/* eslint-env jest */

import { renderString } from '../../utils/nunjucks'

const render = (template, context = mockContext, parse) => renderString(template, context, parse)
const mockContext = {}

describe('Nunjucks filter `render()`', () => {
  it('should render template', () => {
    expect(render(`{{ '{{ 1 + 3 + 5 }} with content'|render() }} and outer content`)).toMatchSnapshot()
  })

  it('should render current context variable', () => {
    expect(render(`
      {% set __globalVar = 'testing __globalVar value' %}

      {{ '{{ __globalVar }} with content'|render() }} and outer content
    `)).toMatchSnapshot()
  })

  it('should render current context macro', () => {
    expect(render(`
      {% macro __MacroTest(value) %}
      <article>Value: {{ value }}</article>
      {% endmacro %}

      {{ "{{ __MacroTest('This is macro value') }} with content"|render()|safe }} and outer content
    `)).toMatchSnapshot()
  })

  it('should render current context macro\'s inner', () => {
    expect(render(`
      {% macro __MacroInnerTest(value) %}
      <article>{{ 1 + 2 + 5 }} and value: {{ value }}</article>
      {% endmacro %}

      {{ __MacroInnerTest('This is macro value')|render()|safe }} and outer content
    `)).toMatchSnapshot()
  })

  it('should render current context macro\'s caller', () => {
    expect(render(`
      {% macro __MacroCallerTest() %}
      <article>Caller value: {{ caller()|render()|safe }}</article>
      {% endmacro %}

      {% call __MacroCallerTest() -%}
      {% raw %}<p>{{ 1 + 2 + 3 }} {{ 'This is caller value' }}</p>{% endraw %}
      {% endcall %}
    `)).toMatchSnapshot()
  })
})
