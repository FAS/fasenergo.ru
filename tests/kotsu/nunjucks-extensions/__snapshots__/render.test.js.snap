// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nunjucks filter \`render()\` should render current context macro 1`] = `
"
      

      
      <article>Value: This is macro value</article>
       with content and outer content
    "
`;

exports[`Nunjucks filter \`render()\` should render current context macro's caller 1`] = `
"
      

      
      <article>Caller value: <p>6 This is caller value</p>
      </article>
      
    "
`;

exports[`Nunjucks filter \`render()\` should render current context macro's inner 1`] = `
"
      

      
      <article>8 and value: This is macro value</article>
       and outer content
    "
`;

exports[`Nunjucks filter \`render()\` should render current context variable 1`] = `
"
      

      testing __globalVar value with content and outer content
    "
`;

exports[`Nunjucks filter \`render()\` should render template 1`] = `"9 with content and outer content"`;
